File: src/app/Foundation.Template.Shell/Services/PermissionService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

using AutoMapper;
using Bones.Flow;

using Foundation.Template.Domain.Models;
using Foundation.Template.Shell.Abstractions;
using Foundation.Template.Shell.Abstractions;
using Foundation.Template.Shell.ViewModels;

using static Foundation.Template.Shell.AutoMapper.Consts;

namespace Foundation.Template.Shell.Services
{
    public class PermissionService : IPermissionService
    {
        private readonly IPermissionProvider _permissionProvider;
        private readonly IQueryHandler<PermissionCategoriesQuery, IEnumerable<PermissionCategory>> _categoriesQueryHandler;
        private readonly IQueryHandler<PermissionsQuery, IEnumerable<PermissionInfos>> _permissionsQueryHandler;
        private readonly IRequestContextProvider _requestContextProvider;
        private readonly IMapper _mapper;

        public PermissionService(
            IQueryHandler<PermissionsQuery, IEnumerable<PermissionInfos>> permissionsQueryHandler,
            IQueryHandler<PermissionCategoriesQuery, IEnumerable<PermissionCategory>> categoriesQueryHandler,
            IRequestContextProvider requestContextProvider,
            IPermissionProvider permissionProvider,
            IMapper mapper
        )
        {
            _categoriesQueryHandler = categoriesQueryHandler;
            _permissionsQueryHandler = permissionsQueryHandler;

            _requestContextProvider = requestContextProvider;
            _permissionProvider = permissionProvider;
            _mapper = mapper;
        }

        public async Task<IEnumerable<string>> GetCurrent()
        {
            var context = _requestContextProvider.Context;
            return await _permissionProvider.GetPermissions(context.OrganisationId.Value, context.ActorId);
        }

        public async Task<IEnumerable<PermissionCategoryViewModel>> GetCategories()
        {
            var query = new PermissionCategoriesQuery();

            var categories = await _categoriesQueryHandler.HandleAsync(query);

            var context = _requestContextProvider.Context;
            return _mapper.Map<IEnumerable<PermissionCategory>, IEnumerable<PermissionCategoryViewModel>>(categories, opt => opt.Items.Add(LANGUAGE, context.LanguageCode));
        }

        public async Task<IEnumerable<PermissionInfosViewModel>> GetMany()
        {
            var query = new PermissionsQuery();

            var result = await _permissionsQueryHandler.HandleAsync(query);

            return _mapper.Map<IEnumerable<PermissionInfos>, IEnumerable<PermissionInfosViewModel>>(result);
        }
    }
}

File: src/app/Foundation.Template.Core-UI/services/permissionService.ts
import axios from "axios";

import { buildURL } from "@dative-gpi/foundation-template-shared-ui";

import { IPermissionService } from '../abstractions';
import { CURRENT_USER_PERMISSIONS_URL, PERMISSIONS_URL, PERMISSION_CATEGORIES_URL } from '../config';
import { PermissionCategory, PermissionCategoryDTO, PermissionInfos, PermissionInfosDTO, PermissionsFilter } from "../domain";


export class PermissionService implements IPermissionService {
	async getCurrent(organisationId: string): Promise<string[]> {
		const response = await axios.get(CURRENT_USER_PERMISSIONS_URL(organisationId));
		const permissions: string[] = response.data;

		return permissions;
	}

	async getMany(organisationId: string, filter: PermissionsFilter): Promise<PermissionInfos[]> {
	  const response = await axios.get(
		buildURL(PERMISSIONS_URL(organisationId), filter)
	  );
	  const dto: PermissionInfosDTO[] = response.data;
  
	  const permissions = dto.map((o) => new PermissionInfos(o));
  
	  return permissions;
	}

	async getCategories(organisationId: string): Promise<PermissionCategory[]> {
		const response = await axios.get(PERMISSION_CATEGORIES_URL(organisationId));
		const dtos: PermissionCategoryDTO[] = response.data;

		const categories = dtos.map(dto => new PermissionCategory(dto));
		return categories;
	}
}